### PR TITLE
sql: re-enable flaky show_trace_nonmetamorphic test

### LINKED ISSUE #58202: sql: audit show_trace opt logic test
`show_trace` opt logic test was skipped because of #43043 (which was closed when a separate issue was fixed). #58160 will unskip the whole test, yet some of the queries will remain skipped. We should audit the file and either delete the skipped queries or make them non-flaky.

### PR DESCRIPTION
**Pull request by cucaroach**
Cleanup some commented out tests

Resolves #58202

Release note: none

### CODE REVIEW

File: pkg/sql/opt/exec/execbuilder/testdata/show_trace_nonmetamorphic [hunk 1]
```diff
@@ -287,35 +274,27 @@ CREATE TABLE t (a INT PRIMARY KEY)
...
-# dist sender send  querying next range at /Table/3/1/12/2/1
-# dist sender send  r6: sending batch 1 Get to (n1,s1):1
-# dist sender send  querying next range at /Table/3/1/1/2/1
-# dist sender send  r6: sending batch 1 Get to (n1,s1):1
-# dist sender send  querying next range at /Table/12/1
-# dist sender send  r8: sending batch 1 Scan to (n1,s1):1
+# This test shows some variation across runs so use statement okay instead
+# of baking in flakey values.
+statement ok
+SELECT DISTINCT node_id, store_id, replica_id
```
Comment (tbg) [inline pkg/sql/opt/exec/execbuilder/testdata/show_trace_nonmetamorphic:280]: Is there a point to this if we're not saying anything about the output?

File: pkg/sql/opt/exec/execbuilder/testdata/show_trace_nonmetamorphic [hunk 2]
```diff
@@ -287,35 +274,27 @@ CREATE TABLE t (a INT PRIMARY KEY)
...
+  FROM [SHOW EXPERIMENTAL_REPLICA TRACE FOR SESSION]
+
+subtest system_table_lookup
+
+# We use AOST to bypass the table cache.
+statement ok
+SET tracing = on,kv; SELECT * FROM system.eventlog AS OF SYSTEM TIME '-1us'; SET tracing = off
+
+query TT
+SELECT operation, regexp_replace(regexp_replace(message, 'job_id:[1-9]\d*', 'job_id:...', 'g'), 'drop_time:\d+', 'drop_time:...', 'g') as message
```
Comment (tbg) [inline pkg/sql/opt/exec/execbuilder/testdata/show_trace_nonmetamorphic:290]: What's with all of the regexp_replace that then doesn't seem to apply to anything in the trace? Kinda feels like this has rotted beyond recognition & purpose.

**Review by tbg — COMMENTED** [review]
Thanks Tommy! I have had this open a few times but never felt confident enough to decide whether these tests are even testing anything anymore, see the inline comments. Feels like you need someone with more skin in the game to make that call (or to guide you towards figuring out what to actually test here!)

**Review by cucaroach — COMMENTED** [review]
I'm from the "every test should carry some load or be deleted" camp.   Happy to nuke these if that's the case.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66001#-:-Me1WpJ6CtjFbbUTGFDX:bw2hzbi)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @tbg)

___
*[pkg/sql/opt/exec/execbuilder/testdata/show_trace_nonmetamorphic, line 280 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66001#-MbCpT-juqHIr6QVI0RZ-r1-280:-Me1VgqZ2VZ-ZihqhS7p:b5b1wcf) ([raw file](https://github.com/cockroachdb/cockroach/blob/6613f4738f8bddca2d89dd0375f0abf6baf379aa/pkg/sql/opt/exec/execbuilder/testdata/show_trace_nonmetamorphic#L280)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

Is there a point to this if we're not saying anything about the output?
</blockquote></details>

The only other test we have for this is sql.TestShowTraceReplica.   If this adds nothing beyond that test I'll nuke it.
___
*[pkg/sql/opt/exec/execbuilder/testdata/show_trace_nonmetamorphic, line 290 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66001#-MbCpT-juqHIr6QVI0RZ-r1-290:-Me1WKw422pJ0kqZ2i3d:b-k0b8sj) ([raw file](https://github.com/cockroachdb/cockroach/blob/6613f4738f8bddca2d89dd0375f0abf6baf379aa/pkg/sql/opt/exec/execbuilder/testdata/show_trace_nonmetamorphic#L290)):*
<details><summary><i>Previously, tbg (Tobias Grieger) wrote…</i></summary><blockquote>

What's with all of the regexp_replace that then doesn't seem to apply to anything in the trace? Kinda feels like this has rotted beyond recognition & purpose.
</blockquote></details>

Looks like some vestigial id redacting, should I just nuke the whole thing or just remove the regexp_replaces?   Feels like we rely on SHOW KV TRACE FOR SESSION for so many other tests now that this test is gratuitous.

**Review by yuzefovich — APPROVED** [review]
nit: update the PR description to close the github issue.

:lgtm:

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66001#-:-MeS966m-Oo3hXKLfNhV:b-5k20zs)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @tbg)

___
*[pkg/sql/opt/exec/execbuilder/testdata/show_trace_nonmetamorphic, line 290 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66001#-MbCpT-juqHIr6QVI0RZ-r1-290:-MeS8m-A1SYt6CbdaaI1:b-objh1u) ([raw file](https://github.com/cockroachdb/cockroach/blob/6613f4738f8bddca2d89dd0375f0abf6baf379aa/pkg/sql/opt/exec/execbuilder/testdata/show_trace_nonmetamorphic#L290)):*
<details><summary><i>Previously, cucaroach (Tommy Reilly) wrote…</i></summary><blockquote>

Looks like some vestigial id redacting, should I just nuke the whole thing or just remove the regexp_replaces?   Feels like we rely on SHOW KV TRACE FOR SESSION for so many other tests now that this test is gratuitous.
</blockquote></details>

I imagine `regexp_replace` part was copy-pasted from somewhere and doesn't seem necessary here.

**Review by cucaroach — COMMENTED** [review]
TFTRs!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66001#-:-Me_OBWM4hTLm1CLX5OQ:bz5mair)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @tbg)

**Pull request merged**